File: Data-Structures-And-Algorithms/Map/leetCode-1-twoSum.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println()
}

func twoSum(nums []int, target int) []int {
	table := map[int]int{}
	for i, x := range nums {
		if p, ok := table[target-x]; ok {
			return []int{p, i}
		}
		table[x] = i
	}
	return nil
}

File: Data-Structures-And-Algorithms/Array/leetCode-1-twoSum.php
class Solution {

    /**
    * @param Integer[] $nums
    * @param Integer $target
    * @return Integer[]
    */
    function twoSum($nums, $target) {
        $map = [];
        foreach ($nums as $k => $v) {
            $map[$v] = $k;
        }
        foreach ($nums as $key => $value) {
            $item = $target - $value;
            if (isset($map[$item]) && $map[$item] != $key) {
                return [$key, $map[$item]];
            }
        }
        return [];
    }

}